src/ftpclass.cc

```diff
@@ -1535,7 +1535,7 @@ int   Ftp::Do()
 	    }
 	 }
 #if USE_SSL
-	 if(strcmp(proxy_auth_type,"joined") || strcmp(proxy_auth_type,"joined-acct"))
+	 if(strcmp(proxy_auth_type,"joined") != 0 && strcmp(proxy_auth_type,"joined-acct") != 0)
```

Comment:
Ok, it's better now. But AUTH cannot also be used for "user" proxy type. Isn't it better to explicitly list allowed proxy types instead? (they are "open" and "proxy-user@host")